src/CppAst/CppClass.cs

```diff
@@ -276,5 +277,14 @@ public override IEnumerable<ICppDeclaration> Children()
                 yield return item;
             }
         }
+
+        public void UpdateTemplateArgument(CppType sourceParam, List<CppTemplateArgument> templateArguments)
+        {
+            // We need to remove all previously registered template arguments corresponding with 'sourceParam'
```

Comment:
Extend this with "...and update it with the new template arguments." to be more clear